File: nuget/ProductShotAI.SiteKit/SiteKit.cs
namespace ProductShotAI.SiteKit;

public static class SiteKit
{
    public const string Base = "https://productshotai.app";
    public const string Brand = "ProductShot AI";
    public const string Description = "AI product photography generator for ecommerce sellers.";

    public static string PageUrl(string path = "/")
    {
        var value = string.IsNullOrWhiteSpace(path) ? "/" : path;
        if (!value.StartsWith("/")) value = "/" + value;
        var clean = value == "/" ? "/" : value.TrimEnd('/') + "/";
        return Base + clean;
    }

    public static string LocalizedUrl(string locale, string path = "/") => locale switch
    {
        "en" => PageUrl(path),
        "zh" or "zh-CN" => PageUrl("/zh" + NormalizeLocalePath(path)),
        _ => throw new ArgumentException($"unsupported locale: {locale}", nameof(locale))
    };

    private static string NormalizeLocalePath(string path)
    {
        var value = string.IsNullOrWhiteSpace(path) ? "/" : path;
        if (!value.StartsWith("/")) value = "/" + value;
        return value == "/" ? "" : value;
    }

    public static string HomeUrl() => PageUrl("/");
    public static string WorkbenchUrl() => $"{Base}/#workbench";
    public static string PricingUrl() => $"{Base}/#pricing";
    public static string BlogUrl() => PageUrl("/blog");
    public static string AboutUrl() => PageUrl("/about");
    public static string ContactUrl() => PageUrl("/contact");
    public static string PrivacyUrl() => PageUrl("/privacy");
    public static string TermsUrl() => PageUrl("/terms");
    public static string RefundPolicyUrl() => PageUrl("/refund-policy");
    public static string ZhHomeUrl() => LocalizedUrl("zh", "/");
}

File: Sources/ProductShotAISiteKit/SiteKit.swift
import Foundation

public enum ProductShotAISiteKit {
    public static let base = "https://productshotai.app"
    public static let brand = "ProductShot AI"
    public static let description = "AI product photography generator for ecommerce sellers."

    public static func pageUrl(_ path: String = "/") -> String {
        var value = path.isEmpty ? "/" : path
        if !value.hasPrefix("/") { value = "/" + value }
        let clean = value == "/" ? "/" : value.trimmingCharacters(in: CharacterSet(charactersIn: "/")) + "/"
        return base + clean
    }

    public static func localizedUrl(_ locale: String, _ path: String = "/") -> String {
        if locale == "en" { return pageUrl(path) }
        if locale == "zh" || locale == "zh-CN" {
            var value = path.isEmpty ? "/" : path
            if !value.hasPrefix("/") { value = "/" + value }
            return pageUrl("/zh" + (value == "/" ? "" : value))
        }
        preconditionFailure("unsupported locale: \(locale)")
    }

    public static func homeUrl() -> String { pageUrl("/") }
    public static func workbenchUrl() -> String { "\(base)/#workbench" }
    public static func pricingUrl() -> String { "\(base)/#pricing" }
    public static func blogUrl() -> String { pageUrl("/blog") }
    public static func contactUrl() -> String { pageUrl("/contact") }
    public static func zhHomeUrl() -> String { localizedUrl("zh", "/") }
}
